library/core/src/ptr/mod.rs

```diff
@@ -1561,6 +1565,171 @@ pub unsafe fn write_volatile<T>(dst: *mut T, src: T) {
 ///
 /// Any questions go to @nagisa.
 #[lang = "align_offset"]
+#[cfg(not(bootstrap))]
+pub(crate) const unsafe fn align_offset<T: Sized>(p: *const T, a: usize) -> usize {
```

Comment:
since this function is never actually called, instead of making its body const fn, you can also just slap `#[rustc_do_not_const_check]` on the function, which ignores its body (we use this for other hooks already)

Reply:
This function *is* actually called, just with a different argument. But turns out the `#[cfg(not(bootstrap))]` wasn't need after all. Calling this function in const eval with the bootstrap compiler will just fail with an error. I've removed most of the duplicated functions now and added comments to explain the different behavior for `cfg(bootstrap)`.